plot_app/configured_plots.py

```diff
@@ -141,7 +141,82 @@ def generate_plots(ulog, px4_ulog, db_data, vehicle_data, link_to_3d_page):
+                        flight_mode_changes = cur_dataset.list_value_changes('nav_state')
+                        flight_mode_changes.append((ulog.last_timestamp, -1))
+                    except (KeyError, IndexError) as error:
+                        flight_mode_changes = []
+                    if flight_mode_changes is None:
+                        flight_mode_changes = []
+                    topic_instance = 0
+                    position_topic_name = 'vehicle_global_position'
+                    for elem in data:
+                        if elem.name == position_topic_name and elem.multi_id == topic_instance:
+                            cur_data = elem
+
+                    pos_lon = cur_data.data['lon']
+                    pos_lat = cur_data.data['lat']
+                    pos_alt = cur_data.data['alt']
+                    pos_t = cur_data.data['timestamp']
+
+                    if 'fix_type' in cur_data.data:
+                        indices = cur_data.data['fix_type'] > 2  # use only data with a fix
+                        pos_lon = pos_lon[indices]
+                        pos_lat = pos_lat[indices]
+                        pos_alt = pos_alt[indices]
+                        pos_t = pos_t[indices]
+
+                    # scale if it's an integer type
+                    lon_type = [f.type_str for f in cur_data.field_data if f.field_name == 'lon']
+                    if len(lon_type) > 0 and lon_type[0] == 'int32_t':
+                        pos_lon = pos_lon / 1e7  # to degrees
+                        pos_lat = pos_lat / 1e7
+                        pos_alt = pos_alt / 1e3  # to meters
+
+                    current_flight_mode = 0
+                    current_flight_mode_idx = 0
+                    if len(flight_mode_changes) > 0:
+                        current_flight_mode = flight_mode_changes[0][1]
+                    last_t = 0
+                    minimum_interval_s = 0.1
+                    pos_data = []
+                    for i in range(len(pos_lon)):
+                        cur_t = pos_t[i]
+                        if(cur_t - last_t) / 1e6 > minimum_interval_s:
+                            pos_data.append([pos_lat[i], pos_lon[i], current_flight_mode, \
+                                rgb_colors(current_flight_mode)])# assume timestamp is in [us]
+                            last_t = cur_t
+                            while current_flight_mode_idx < len(flight_mode_changes) - 1 and \
+                                    flight_mode_changes[current_flight_mode_idx + 1][0] <= cur_t:
+                                current_flight_mode_idx += 1
+                                current_flight_mode = flight_mode_changes[current_flight_mode_idx][1]
+                    curdoc().template_variables['pos_polyline'] = pos_data
+
+                ulog_to_polyline(ulog)
+            except:
+                pass
+            if not is_running_locally(): # Leaflet Map
```

Comment:
Can you remove this line?

Reply:
the same, so I think we should use leaflet both in remote(server) mode and local mode, is that right?

Reply:
Yes, exactly.

Reply:
understand！